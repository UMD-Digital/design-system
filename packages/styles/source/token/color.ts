/**
 * @module token/color
 * Provides color tokens for the design system.
 */

export type ColorScale = typeof colorScale;
export type BaseColors = typeof baseColors;

/**
 * Color scale with various shades for different colors.
 * @returns {ColorScale} Color scale object with various color categories and shades.
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * Styles.token.colorScale.gray[500]
 * ```
 * @since 1.1.0
 */
export const colorScale = {
  gray: {
    50: '#FAFAFA',
    100: '#F1F1F1',
    200: '#E6E6E6',
    400: '#7F7F7F',
    500: '#757575',
    700: '#454545',
    900: '#242424',
  },
  red: {
    500: '#E21833',
    900: '#A90007',
  },
  blue: {
    500: '#2F7EDA',
  },
  green: {
    500: '#008000',
  },
  gold: {
    500: '#FFD200',
  },
} as const;

/**
 * Base colors for black and white.
 * @returns {BaseColors} Base colors object with white and black.
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * Styles.token.baseColors.white
 * ```
 * @since 1.1.0
 */
export const baseColors = {
  white: '#FFFFFF',
  black: '#000000',
} as const;

/**
 * Default color tokens for the design system.
 * @returns {object} Color tokens with semantic naming.
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * Styles.token.color.red
 * ```
 * @since 1.1.0
 */
export default {
  red: colorScale.red[500],
  blue: colorScale.blue[500],
  green: colorScale.green[500],
  gold: colorScale.gold[500],
  white: baseColors.white,
  black: baseColors.black,
  redDark: colorScale.red[900],
  gray: {
    darker: colorScale.gray[900],
    dark: colorScale.gray[700],
    mediumAA: colorScale.gray[500],
    medium: colorScale.gray[400],
    light: colorScale.gray[200],
    lighter: colorScale.gray[100],
    lightest: colorScale.gray[50],
  },
};
