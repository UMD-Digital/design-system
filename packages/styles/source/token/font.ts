/**
 * @module token/font
 * Provides font-related design tokens including sizes, weights, and font families.
 */

/**
 * Font size tokens.
 * @type {Object<string, string>}
 * @property {string} min - 12px font size
 * @property {string} sm - 14px font size
 * @property {string} base - 16px font size (default)
 * @property {string} lg - 18px font size
 * @property {string} xl - 20px font size
 * @property {string} 2xl - 22px font size
 * @property {string} 3xl - 24px font size
 * @property {string} 4xl - 32px font size
 * @property {string} 5xl - 48px font size
 * @property {string} 6xl - 56px font size
 * @property {string} 7xl - 64px font size
 * @property {string} 8xl - 72px font size
 * @property {string} 9xl - 80px font size
 * @property {string} 10xl - 96px font size
 * @property {string} max - 120px font size
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * Styles.token.font.size.base // '16px'
 * ```
 * @since 1.8.0
 */
export const size = {
  min: '12px',
  sm: '14px',
  base: '16px',
  lg: '18px',
  xl: '20px',
  '2xl': '22px',
  '3xl': '24px',
  '4xl': '32px',
  '5xl': '48px',
  '6xl': '56px',
  '7xl': '64px',
  '8xl': '72px',
  '9xl': '80px',
  '10xl': '96px',
  max: '120px',
};

/**
 * Font weight tokens.
 * @type {Object<string, string>}
 * @property {string} thin - 100 weight
 * @property {string} extraLight - 200 weight
 * @property {string} light - 300 weight
 * @property {string} normal - 400 weight (default)
 * @property {string} medium - 500 weight
 * @property {string} semiBold - 600 weight
 * @property {string} bold - 700 weight
 * @property {string} extraBold - 800 weight
 * @property {string} black - 900 weight
 * @property {string} extraBlack - 950 weight
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * Styles.token.font.weight.bold // '700'
 * ```
 * @since 1.8.0
 */
export const weight = {
  thin: '100',
  extraLight: '200',
  light: '300',
  normal: '400',
  medium: '500',
  semiBold: '600',
  bold: '700',
  extraBold: '800',
  black: '900',
  extraBlack: '950',
};

/**
 * Font family tokens.
 * @type {Object<string, string>}
 * @property {string} campaign - Barlow Condensed font stack for campaign/marketing text
 * @property {string} sans - Interstate font stack for primary sans-serif text
 * @property {string} serif - Crimson Pro font stack for serif text
 * @property {string} mono - Monospace font stack for code/technical text
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * Styles.token.font.family.sans // "'Interstate', Helvetica, Arial, Verdana, sans-serif"
 * ```
 * @since 1.8.0
 */
export const family = {
  campaign: "'Barlow Condensed', Arial Narrow, sans-serif",
  sans: "'Interstate', Helvetica, Arial, Verdana, sans-serif",
  serif: "'Crimson Pro', Georgia, serif",
  mono: 'monospace',
};
