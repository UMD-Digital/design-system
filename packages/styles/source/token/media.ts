/**
 * @module token/media
 * Provides responsive design tokens for breakpoints and media queries.
 */

/**
 * Numeric breakpoint values in pixels.
 * @type {Object<string, {min: number, max?: number}>}
 * @property {Object} small - Small screen breakpoint (320px-479px)
 * @property {Object} medium - Medium screen breakpoint (480px-649px)
 * @property {Object} large - Large screen breakpoint (650px-767px) 
 * @property {Object} tablet - Tablet breakpoint (768px-1023px)
 * @property {Object} desktop - Desktop breakpoint (1024px-1199px)
 * @property {Object} highDef - High-definition screen breakpoint (1200px-1499px)
 * @property {Object} maximum - Maximum screen size breakpoint (1500px+)
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * Styles.token.media.breakpointValues.desktop.min // 1024
 * ```
 * @since 1.8.0
 */
export const breakpointValues = {
  small: { min: 320, max: 479 },
  medium: { min: 480, max: 649 },
  large: { min: 650, max: 767 },
  tablet: { min: 768, max: 1023 },
  desktop: { min: 1024, max: 1199 },
  highDef: { min: 1200, max: 1499 },
  maximum: { min: 1500 },
};

/**
 * Breakpoint values with 'px' units for CSS usage.
 * @type {Object<string, {min: string, max?: string}>}
 * @property {Object} small - Small screen breakpoint strings ("320px"-"479px")
 * @property {Object} medium - Medium screen breakpoint strings ("480px"-"649px")
 * @property {Object} large - Large screen breakpoint strings ("650px"-"767px")
 * @property {Object} tablet - Tablet breakpoint strings ("768px"-"1023px")
 * @property {Object} desktop - Desktop breakpoint strings ("1024px"-"1199px")
 * @property {Object} highDef - High-definition screen breakpoint strings ("1200px"-"1499px")
 * @property {Object} maximum - Maximum screen size breakpoint string ("1500px"+)
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * Styles.token.media.breakpoints.desktop.min // "1024px"
 * ```
 * @since 1.8.0
 */
export const breakpoints = {
  small: {
    min: `${breakpointValues.small.min}px`,
    max: `${breakpointValues.small.max}px`,
  },
  medium: {
    min: `${breakpointValues.medium.min}px`,
    max: `${breakpointValues.medium.max}px`,
  },
  large: {
    min: `${breakpointValues.large.min}px`,
    max: `${breakpointValues.large.max}px`,
  },
  tablet: {
    min: `${breakpointValues.tablet.min}px`,
    max: `${breakpointValues.tablet.max}px`,
  },
  desktop: {
    min: `${breakpointValues.desktop.min}px`,
    max: `${breakpointValues.desktop.max}px`,
  },
  highDef: {
    min: `${breakpointValues.highDef.min}px`,
    max: `${breakpointValues.highDef.max}px`,
  },
  maximum: { min: `${breakpointValues.maximum.min}px` },
};

/**
 * Complete media query strings ready for use in CSS.
 * @type {Object<string, {min: string, max?: string}>}
 * @property {Object} small - Small screen media queries
 * @property {Object} medium - Medium screen media queries
 * @property {Object} large - Large screen media queries
 * @property {Object} tablet - Tablet media queries
 * @property {Object} desktop - Desktop media queries
 * @property {Object} highDef - High-definition screen media queries
 * @property {Object} maximum - Maximum screen size media query
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * // Use in JSS/CSS-in-JS
 * const styles = {
 *   [`@media (${Styles.token.media.queries.desktop.min})`]: {
 *     // Desktop-specific styles
 *   }
 * };
 * ```
 * @since 1.8.0
 */
export const queries = {
  small: {
    min: `min-width: ${breakpoints.small.min}`,
    max: `max-width: ${breakpoints.small.max}`,
  },
  medium: {
    min: `min-width: ${breakpoints.medium.min}`,
    max: `max-width: ${breakpoints.medium.max}`,
  },
  large: {
    min: `min-width: ${breakpoints.large.min}`,
    max: `max-width: ${breakpoints.large.max}`,
  },
  tablet: {
    min: `min-width: ${breakpoints.tablet.min}`,
    max: `max-width: ${breakpoints.tablet.max}`,
  },
  desktop: {
    min: `min-width: ${breakpoints.desktop.min}`,
    max: `max-width: ${breakpoints.desktop.max}`,
  },
  highDef: {
    min: `min-width: ${breakpoints.highDef.min}`,
    max: `max-width: ${breakpoints.highDef.max}`,
  },
  maximum: { min: `min-width: ${breakpoints.maximum.min}` },
};
