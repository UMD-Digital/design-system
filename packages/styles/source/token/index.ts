/**
 * @module token
 * Design tokens and system variables.
 */

import color from './color';
import spacing from './spacing';

/**
 * Media query breakpoints and utilities.
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * Styles.token.media
 * ```
 * @since 1.1.0
 */
export * as media from './media';

/**
 * Typography and font tokens.
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * Styles.token.font
 * ```
 * @since 1.1.0
 */
export * as font from './font';

/**
 * Color system tokens.
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * Styles.token.color
 * ```
 * @since 1.1.0
 */
export { color };

/**
 * Spacing system tokens.
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * Styles.token.spacing
 * ```
 * @since 1.1.0
 */
export { spacing };
