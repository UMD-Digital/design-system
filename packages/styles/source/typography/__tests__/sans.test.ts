import * as sans from '../sans';
import { font, media } from '../../token';

describe('typography/sans', () => {
  it('should match snapshot', () => {
    expect(sans).toMatchSnapshot();
  });

  describe('size objects', () => {
    it('should have correctly defined size values', () => {
      // Test individual size exports
      expect(sans.largest).toMatchSnapshot();
      expect(sans.extraLarge).toMatchSnapshot();
      expect(sans.larger).toMatchSnapshot();
      expect(sans.large).toMatchSnapshot();
      expect(sans.medium).toMatchSnapshot();
      expect(sans.small).toMatchSnapshot();
      expect(sans.smaller).toMatchSnapshot();
      expect(sans.min).toMatchSnapshot();
    });

    it('should use the correct font family for all sizes', () => {
      // All sizes should use sans font family
      [
        sans.largest,
        sans.extraLarge,
        sans.larger,
        sans.large,
        sans.medium,
        sans.small,
        sans.smaller,
        sans.min,
      ].forEach((size) => {
        expect(size.fontFamily).toBe(font.family.sans);
      });
    });
  });

  describe('fonts objects', () => {
    it('should have all expected font objects with classNames', () => {
      // Test that all fonts have class names
      expect(sans.fonts.largest.className).toBe('umd-sans-largest');
      expect(sans.fonts.extraLarge.className).toBe('umd-sans-extralarge');
      expect(sans.fonts.larger.className).toBe('umd-sans-larger');
      expect(sans.fonts.large.className).toBe('umd-sans-large');
      expect(sans.fonts.medium.className).toBe('umd-sans-medium');
      expect(sans.fonts.small.className).toBe('umd-sans-small');
      expect(sans.fonts.smaller.className).toBe('umd-sans-smaller');
      expect(sans.fonts.min.className).toBe('umd-sans-min');
    });

    it('should match snapshot for all fonts objects', () => {
      expect(sans.fonts).toMatchSnapshot();
    });
  });

  describe('scaling fonts', () => {
    it('should match snapshot for scaling fonts', () => {
      expect(sans.scalingFonts).toMatchSnapshot();
    });
  });

  describe('transformations', () => {
    it('should have correctly defined transformations', () => {
      expect(sans.transformations.largestUppercase.fontWeight).toBe(
        font.weight.extraBold,
      );
      expect(sans.transformations.largestUppercase.textTransform).toBe(
        'uppercase',
      );

      expect(sans.transformations.extraLargeUppercase.fontWeight).toBe(
        font.weight.extraBold,
      );
      expect(sans.transformations.extraLargeUppercase.textTransform).toBe(
        'uppercase',
      );

      expect(sans.transformations.extraLargeBold.fontWeight).toBe(
        font.weight.bold,
      );
      expect(sans.transformations.largerBold.fontWeight).toBe(font.weight.bold);
    });

    it('should match snapshot for all transformations', () => {
      expect(sans.transformations).toMatchSnapshot();
    });
  });
});
