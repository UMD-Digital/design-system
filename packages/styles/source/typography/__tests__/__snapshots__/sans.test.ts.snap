// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`typography/sans fonts objects should match snapshot for all fonts objects 1`] = `
{
  "extraLarge": {
    "@media (min-width: 1024px)": {
      "fontSize": "32px",
      "lineHeight": "1.125em",
    },
    "@media (min-width: 650px)": {
      "fontSize": "calc(18px + 1.16vw)",
    },
    "className": "umd-sans-extralarge",
    "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
    "fontSize": "18px",
    "lineHeight": "1.25em",
  },
  "large": {
    "className": "umd-sans-large",
    "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
    "fontSize": "18px",
    "fontWeight": "700",
    "lineHeight": "1.25em",
  },
  "larger": {
    "@media (min-width: 1024px)": {
      "fontSize": "22px",
      "lineHeight": "1.25em",
    },
    "@media (min-width: 650px)": {
      "fontSize": "calc(18px + 0.5vw)",
    },
    "className": "umd-sans-larger",
    "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
    "fontSize": "18px",
    "lineHeight": "1.40em",
  },
  "largest": {
    "@media (min-width: 1024px)": {
      "fontSize": "48px",
      "lineHeight": "1.04em",
    },
    "@media (min-width: 650px)": {
      "fontSize": "calc(24px + 2vw)",
    },
    "className": "umd-sans-largest",
    "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
    "fontSize": "22px",
    "fontWeight": "700",
    "lineHeight": "1.25em",
  },
  "maximum": {
    "className": "umd-sans-maximum",
    "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
    "fontSize": "32px",
    "fontWeight": "700",
    "lineHeight": "1.125em",
  },
  "medium": {
    "@media (min-width: 1024px)": {
      "fontSize": "18px",
      "lineHeight": "1.55em",
    },
    "@media (min-width: 650px)": {
      "fontSize": "calc(16px + 0.16vw)",
    },
    "className": "umd-sans-medium",
    "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
    "fontSize": "16px",
    "lineHeight": "1.375em",
  },
  "min": {
    "className": "umd-sans-min",
    "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
    "fontSize": "12px",
    "lineHeight": "1.16em",
  },
  "small": {
    "@media (min-width: 1024px)": {
      "fontSize": "16px",
      "lineHeight": "1.375em",
    },
    "@media (min-width: 650px)": {
      "fontSize": "calc(14px + 0.16vw)",
    },
    "className": "umd-sans-small",
    "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
    "fontSize": "16px",
    "lineHeight": "1.375em",
  },
  "smaller": {
    "@media (min-width: 1024px)": {
      "fontSize": "14px",
      "lineHeight": "1.28em",
    },
    "@media (min-width: 650px)": {
      "fontSize": "calc(12px + 0.16vw)",
    },
    "className": "umd-sans-smaller",
    "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
    "fontSize": "14px",
    "lineHeight": "1.28em",
  },
}
`;

exports[`typography/sans scaling fonts should match snapshot for scaling fonts 1`] = `
{
  "larger": {
    "@container (min-width: 480px)": {
      "fontSize": "32px",
    },
    "className": "umd-sans-scaling-larger",
    "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
    "fontSize": "18px",
    "lineHeight": "1.25em",
  },
  "min": {
    "@container (min-width: 480px)": {
      "@media (min-width: 1024px)": {
        "fontSize": "16px",
        "lineHeight": "1.375em",
      },
      "@media (min-width: 650px)": {
        "fontSize": "calc(14px + 0.16vw)",
      },
      "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
      "fontSize": "16px",
      "lineHeight": "1.375em",
    },
    "className": "umd-sans-scaling-min",
    "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
    "fontSize": "12px",
    "lineHeight": "1.16em",
  },
}
`;

exports[`typography/sans should match snapshot 1`] = `
{
  "extraLarge": {
    "@media (min-width: 1024px)": {
      "fontSize": "32px",
      "lineHeight": "1.125em",
    },
    "@media (min-width: 650px)": {
      "fontSize": "calc(18px + 1.16vw)",
    },
    "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
    "fontSize": "18px",
    "lineHeight": "1.25em",
  },
  "fonts": {
    "extraLarge": {
      "@media (min-width: 1024px)": {
        "fontSize": "32px",
        "lineHeight": "1.125em",
      },
      "@media (min-width: 650px)": {
        "fontSize": "calc(18px + 1.16vw)",
      },
      "className": "umd-sans-extralarge",
      "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
      "fontSize": "18px",
      "lineHeight": "1.25em",
    },
    "large": {
      "className": "umd-sans-large",
      "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
      "fontSize": "18px",
      "fontWeight": "700",
      "lineHeight": "1.25em",
    },
    "larger": {
      "@media (min-width: 1024px)": {
        "fontSize": "22px",
        "lineHeight": "1.25em",
      },
      "@media (min-width: 650px)": {
        "fontSize": "calc(18px + 0.5vw)",
      },
      "className": "umd-sans-larger",
      "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
      "fontSize": "18px",
      "lineHeight": "1.40em",
    },
    "largest": {
      "@media (min-width: 1024px)": {
        "fontSize": "48px",
        "lineHeight": "1.04em",
      },
      "@media (min-width: 650px)": {
        "fontSize": "calc(24px + 2vw)",
      },
      "className": "umd-sans-largest",
      "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
      "fontSize": "22px",
      "fontWeight": "700",
      "lineHeight": "1.25em",
    },
    "maximum": {
      "className": "umd-sans-maximum",
      "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
      "fontSize": "32px",
      "fontWeight": "700",
      "lineHeight": "1.125em",
    },
    "medium": {
      "@media (min-width: 1024px)": {
        "fontSize": "18px",
        "lineHeight": "1.55em",
      },
      "@media (min-width: 650px)": {
        "fontSize": "calc(16px + 0.16vw)",
      },
      "className": "umd-sans-medium",
      "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
      "fontSize": "16px",
      "lineHeight": "1.375em",
    },
    "min": {
      "className": "umd-sans-min",
      "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
      "fontSize": "12px",
      "lineHeight": "1.16em",
    },
    "small": {
      "@media (min-width: 1024px)": {
        "fontSize": "16px",
        "lineHeight": "1.375em",
      },
      "@media (min-width: 650px)": {
        "fontSize": "calc(14px + 0.16vw)",
      },
      "className": "umd-sans-small",
      "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
      "fontSize": "16px",
      "lineHeight": "1.375em",
    },
    "smaller": {
      "@media (min-width: 1024px)": {
        "fontSize": "14px",
        "lineHeight": "1.28em",
      },
      "@media (min-width: 650px)": {
        "fontSize": "calc(12px + 0.16vw)",
      },
      "className": "umd-sans-smaller",
      "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
      "fontSize": "14px",
      "lineHeight": "1.28em",
    },
  },
  "large": {
    "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
    "fontSize": "18px",
    "fontWeight": "700",
    "lineHeight": "1.25em",
  },
  "larger": {
    "@media (min-width: 1024px)": {
      "fontSize": "22px",
      "lineHeight": "1.25em",
    },
    "@media (min-width: 650px)": {
      "fontSize": "calc(18px + 0.5vw)",
    },
    "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
    "fontSize": "18px",
    "lineHeight": "1.40em",
  },
  "largest": {
    "@media (min-width: 1024px)": {
      "fontSize": "48px",
      "lineHeight": "1.04em",
    },
    "@media (min-width: 650px)": {
      "fontSize": "calc(24px + 2vw)",
    },
    "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
    "fontSize": "22px",
    "fontWeight": "700",
    "lineHeight": "1.25em",
  },
  "maxium": {
    "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
    "fontSize": "32px",
    "fontWeight": "700",
    "lineHeight": "1.125em",
  },
  "medium": {
    "@media (min-width: 1024px)": {
      "fontSize": "18px",
      "lineHeight": "1.55em",
    },
    "@media (min-width: 650px)": {
      "fontSize": "calc(16px + 0.16vw)",
    },
    "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
    "fontSize": "16px",
    "lineHeight": "1.375em",
  },
  "min": {
    "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
    "fontSize": "12px",
    "lineHeight": "1.16em",
  },
  "scalingFonts": {
    "larger": {
      "@container (min-width: 480px)": {
        "fontSize": "32px",
      },
      "className": "umd-sans-scaling-larger",
      "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
      "fontSize": "18px",
      "lineHeight": "1.25em",
    },
    "min": {
      "@container (min-width: 480px)": {
        "@media (min-width: 1024px)": {
          "fontSize": "16px",
          "lineHeight": "1.375em",
        },
        "@media (min-width: 650px)": {
          "fontSize": "calc(14px + 0.16vw)",
        },
        "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
        "fontSize": "16px",
        "lineHeight": "1.375em",
      },
      "className": "umd-sans-scaling-min",
      "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
      "fontSize": "12px",
      "lineHeight": "1.16em",
    },
  },
  "scalingLarger": {
    "@container (min-width: 480px)": {
      "fontSize": "32px",
    },
    "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
    "fontSize": "18px",
    "lineHeight": "1.25em",
  },
  "scalingMin": {
    "@container (min-width: 480px)": {
      "@media (min-width: 1024px)": {
        "fontSize": "16px",
        "lineHeight": "1.375em",
      },
      "@media (min-width: 650px)": {
        "fontSize": "calc(14px + 0.16vw)",
      },
      "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
      "fontSize": "16px",
      "lineHeight": "1.375em",
    },
    "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
    "fontSize": "12px",
    "lineHeight": "1.16em",
  },
  "small": {
    "@media (min-width: 1024px)": {
      "fontSize": "16px",
      "lineHeight": "1.375em",
    },
    "@media (min-width: 650px)": {
      "fontSize": "calc(14px + 0.16vw)",
    },
    "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
    "fontSize": "16px",
    "lineHeight": "1.375em",
  },
  "smaller": {
    "@media (min-width: 1024px)": {
      "fontSize": "14px",
      "lineHeight": "1.28em",
    },
    "@media (min-width: 650px)": {
      "fontSize": "calc(12px + 0.16vw)",
    },
    "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
    "fontSize": "14px",
    "lineHeight": "1.28em",
  },
  "transformations": {
    "extraLargeBold": {
      "@media (min-width: 1024px)": {
        "fontSize": "32px",
        "lineHeight": "1.125em",
      },
      "@media (min-width: 650px)": {
        "fontSize": "calc(18px + 1.16vw)",
      },
      "className": "umd-sans-extralarge-bold",
      "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
      "fontSize": "18px",
      "fontWeight": "700",
      "lineHeight": "1.25em",
    },
    "extraLargeUppercase": {
      "@media (min-width: 1024px)": {
        "fontSize": "32px",
        "lineHeight": "1.125em",
      },
      "@media (min-width: 650px)": {
        "fontSize": "calc(18px + 1.16vw)",
      },
      "className": "umd-sans-extralarge-uppercase",
      "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
      "fontSize": "18px",
      "fontWeight": "800",
      "lineHeight": "1.25em",
      "textTransform": "uppercase",
    },
    "largerBold": {
      "@media (min-width: 1024px)": {
        "fontSize": "22px",
        "lineHeight": "1.25em",
      },
      "@media (min-width: 650px)": {
        "fontSize": "calc(18px + 0.5vw)",
      },
      "className": "umd-sans-larger-bold",
      "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
      "fontSize": "18px",
      "fontWeight": "700",
      "lineHeight": "1.40em",
    },
    "largestUppercase": {
      "@media (min-width: 1024px)": {
        "fontSize": "48px",
        "lineHeight": "1.04em",
      },
      "@media (min-width: 650px)": {
        "fontSize": "calc(24px + 2vw)",
      },
      "className": "umd-sans-largest-uppercase",
      "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
      "fontSize": "22px",
      "fontWeight": "800",
      "lineHeight": "1.25em",
      "textTransform": "uppercase",
    },
  },
}
`;

exports[`typography/sans size objects should have correctly defined size values 1`] = `
{
  "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
  "fontSize": "32px",
  "fontWeight": "700",
  "lineHeight": "1.125em",
}
`;

exports[`typography/sans size objects should have correctly defined size values 2`] = `
{
  "@media (min-width: 1024px)": {
    "fontSize": "48px",
    "lineHeight": "1.04em",
  },
  "@media (min-width: 650px)": {
    "fontSize": "calc(24px + 2vw)",
  },
  "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
  "fontSize": "22px",
  "fontWeight": "700",
  "lineHeight": "1.25em",
}
`;

exports[`typography/sans size objects should have correctly defined size values 3`] = `
{
  "@media (min-width: 1024px)": {
    "fontSize": "32px",
    "lineHeight": "1.125em",
  },
  "@media (min-width: 650px)": {
    "fontSize": "calc(18px + 1.16vw)",
  },
  "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
  "fontSize": "18px",
  "lineHeight": "1.25em",
}
`;

exports[`typography/sans size objects should have correctly defined size values 4`] = `
{
  "@media (min-width: 1024px)": {
    "fontSize": "22px",
    "lineHeight": "1.25em",
  },
  "@media (min-width: 650px)": {
    "fontSize": "calc(18px + 0.5vw)",
  },
  "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
  "fontSize": "18px",
  "lineHeight": "1.40em",
}
`;

exports[`typography/sans size objects should have correctly defined size values 5`] = `
{
  "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
  "fontSize": "18px",
  "fontWeight": "700",
  "lineHeight": "1.25em",
}
`;

exports[`typography/sans size objects should have correctly defined size values 6`] = `
{
  "@media (min-width: 1024px)": {
    "fontSize": "18px",
    "lineHeight": "1.55em",
  },
  "@media (min-width: 650px)": {
    "fontSize": "calc(16px + 0.16vw)",
  },
  "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
  "fontSize": "16px",
  "lineHeight": "1.375em",
}
`;

exports[`typography/sans size objects should have correctly defined size values 7`] = `
{
  "@media (min-width: 1024px)": {
    "fontSize": "16px",
    "lineHeight": "1.375em",
  },
  "@media (min-width: 650px)": {
    "fontSize": "calc(14px + 0.16vw)",
  },
  "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
  "fontSize": "16px",
  "lineHeight": "1.375em",
}
`;

exports[`typography/sans size objects should have correctly defined size values 8`] = `
{
  "@media (min-width: 1024px)": {
    "fontSize": "14px",
    "lineHeight": "1.28em",
  },
  "@media (min-width: 650px)": {
    "fontSize": "calc(12px + 0.16vw)",
  },
  "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
  "fontSize": "14px",
  "lineHeight": "1.28em",
}
`;

exports[`typography/sans size objects should have correctly defined size values 9`] = `
{
  "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
  "fontSize": "12px",
  "lineHeight": "1.16em",
}
`;

exports[`typography/sans transformations should match snapshot for all transformations 1`] = `
{
  "extraLargeBold": {
    "@media (min-width: 1024px)": {
      "fontSize": "32px",
      "lineHeight": "1.125em",
    },
    "@media (min-width: 650px)": {
      "fontSize": "calc(18px + 1.16vw)",
    },
    "className": "umd-sans-extralarge-bold",
    "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
    "fontSize": "18px",
    "fontWeight": "700",
    "lineHeight": "1.25em",
  },
  "extraLargeUppercase": {
    "@media (min-width: 1024px)": {
      "fontSize": "32px",
      "lineHeight": "1.125em",
    },
    "@media (min-width: 650px)": {
      "fontSize": "calc(18px + 1.16vw)",
    },
    "className": "umd-sans-extralarge-uppercase",
    "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
    "fontSize": "18px",
    "fontWeight": "800",
    "lineHeight": "1.25em",
    "textTransform": "uppercase",
  },
  "largerBold": {
    "@media (min-width: 1024px)": {
      "fontSize": "22px",
      "lineHeight": "1.25em",
    },
    "@media (min-width: 650px)": {
      "fontSize": "calc(18px + 0.5vw)",
    },
    "className": "umd-sans-larger-bold",
    "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
    "fontSize": "18px",
    "fontWeight": "700",
    "lineHeight": "1.40em",
  },
  "largestUppercase": {
    "@media (min-width: 1024px)": {
      "fontSize": "48px",
      "lineHeight": "1.04em",
    },
    "@media (min-width: 650px)": {
      "fontSize": "calc(24px + 2vw)",
    },
    "className": "umd-sans-largest-uppercase",
    "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
    "fontSize": "22px",
    "fontWeight": "800",
    "lineHeight": "1.25em",
    "textTransform": "uppercase",
  },
}
`;
